src/ControlSystem/ExpirationTimes.hpp

```
@@ -0,0 +1,135 @@
+ * function of time expiration time, our measurement timescales can't be valid
+ * and we must wait for updated ones. To achieve this, we set the measurement
+ * expiration time *before* the function of time expiration time, but *after*
+ * the previous measurement (the update measurement). The factor of one half is
+ * just to guarantee we are more than epsilon before the function of time
+ * expiration time and more than epsilon after the update measurement.
+ */
+double measurement_expiration_time(const double time,
+                                   const DataVector& old_measurement_timescales,
+                                   const DataVector& new_measurement_timescales,
+                                   const int measurements_per_update);
+
+/*!
+ * \ingroup ControlSystemGroup
+ * \brief Construct the initial expiration times for functions of time that are
+ * controlled by a control system
+ *
+ * The expiration times are constructed using inputs from control system
+ * OptionHolders as an unordered map from the name of the function of time being
+ * controlled to the expiration time. The expiration time is computed as
+ * \f$\tau_\mathrm{exp} = \alpha_\mathrm{update} \tau_\mathrm{damp}\f$ where
+ * \f$\alpha_\mathrm{update}\f$ is the update fraction supplied as input to the
+ * Controller and \f$\tau_\mathrm{damp}\f$ is/are the damping timescales
+ * supplied from the TimescaleTuner (\f$\tau_\mathrm{damp}\f$ is a DataVector
+ * with as many components as the corresponding function of time, thus
+ * \f$\tau_\mathrm{exp}\f$ will also be a DataVector of the same length).
+ *
+ * If the control system isn't active then expiration time is
+ * `std::numeric_limits<double>::infinity()`.
+ *
+ * To protect against bad inputs, if the initial expiration time that is
+ * calculated is smaller than the initial time step, then the expiration time is
+ * simply set to the initial time step. However, the MeasurementTimescales have
```

Given this caveat and semi-silent error if both are invalid, would it be better to error on bad inputs? Or set the initial measurement time to 1/2 the initial time step?